src/main/resources/META-INF/rewrite/java-version-17.yml

```diff
@@ -47,3 +49,41 @@ recipeList:
       version: 17
   - org.openrewrite.java.migrate.maven.UseMavenCompilerPluginReleaseConfiguration:
       releaseVersion: 17
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.DeprecatedJavaxSecurityCert
+displayName: Use javax.security.cert
+description: The javax.security.cert package has been deprecated for removal
+tags:
+  - java17
+recipeList:
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: javax.security.cert
+      newFullyQualifiedTypeName: java.security.cert
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.RemovedLegacySunJSSEProviderName
+displayName: Use "SunJSSE"
+description: The com.sun.net.ssl.internal.ssl.Provider provider name was removed
+tags:
+  - java17
+recipeList:
+  - org.openrewrite.java.ChangeStringLiteral:
+      oldFullyQualifiedTypeName: com\.sun\.net\.ssl\.internal\.ssl\.Provider
+      newFullyQualifiedTypeName: SunJSSE
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.DeprecatedLogRecordMethods
+displayName: Use getLongThreadID and setLongThreadID
+description: Avoid using the deprecated methods in java.util.logging.LogRecord
+tags:
+  - java17
+recipeList:
+  - org.openrewrite.java.ChangeMethodName:
+      methodPattern: java.util.logging.LogRecord getThreadID()
+      newMethodName: getLongThreadID
```

Comment:
The old method returned an `int`, whereas the new method returns a `long`. If the return value is assigned to an `int` field or variable, or passed into a method that accepts an `int`, we might need to change the type to `long`, or the compiler won't allow the code change.

Reply:
You're right. I overlooked the type of the value getting returned. Is it possible to account for that in the YAML, or would this need to change to a Java recipe?

Reply:
That would have to be a custom recipe I'm afraid. Likely find any assignments to fields and variables first, as that's what I think is most likely, and methods we can tackle separately. Perhaps best to take that one on in a separate PR, such that we can merge the other two recipes here early.
`setLongThreadID` would have no such issues, as we can upcast from `int` to `long`.

Reply:
I've restored the `setThreadID`; don't think it's very common, but we can trivially cover that already.
https://github.com/openrewrite/rewrite-migrate-java/pull/290/commits/36f8490f0e827673fa5d8329f5012a106230e0de